app/utils/BFS.js

```diff
@@ -0,0 +1,122 @@
+const pathFind = (field, startP, endP) => {
+	const path = [];
+
+	// tests if the point is inside the tested field				
+	var inField = function (point) {
+		if (point[0]<field.length && point[0] >=0) {
+			if (point[1]<field[0].length && point[1] >=0) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	// tests if the point was already
+	var visited = function (point) {
+		if (field[point[0]][point[1]].stepVisited<0) {
+			return false;
+		}
+		return true;
+	}
+
+	// tests if a certain step is allowed
+	var allowedStep = function (point) {
+		if (field[point[1]][point[0]].allowed) {
+			return true;
+		} else if (!field[point[0]][point[1]].allowed) {
+			return false;	
+		}
+	}
+
+	// tests if the tested point is in fact the ending point of the search
+	var isEndPoint = function(point) {
+		if (point === endP) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	// returns the neighbours of a certain point
```

Comment:
Comments don't help. Name your identifiers so you won't need comments: `getPointNeighbours`